### ISSUE TITLE
Remove arbitrary caps on charges available for crafting in vehicle part/appliance pseudo tools

### ISSUE THREAD

**Issue by AmarinReyny** [CleverRaven/Cataclysm-DDA#63628]
### Is your feature request related to a problem? Please describe.

Vehicle part/appliance pseudo tools seem to have an arbitrary cap on the number of charges that can be used for crafting, which does not seem to be at all related to the actual number of charges held in batteries connected to the vehicle part of appliance which provides the pseudo tools in question. This can cause problems when it comes to batch crafting, especially with recipes that have a much higher charge cost than most other recipes.

It can also be a huge disappointment to players who did all the work to find the necessary components and to all the work to build a vehicle with multiple very large vehicle batteries - or have added very large batteries to every bit of space we don't regularly use in our basecamps in order to store the power for our electric grids - just so they can have the available charges necessary to batch craft large amounts of items with recipes calling for a ludicrous amount of charges, and even people who (like myself) have added numerous recipes into basecamp recipe groups via json so we can have our numerous followers take care of more tedious recipes.

### Solution you would like.

I'd like for the amount of available charges for crafting with vehicle part and appliance pseudo tools to have no cap other than how many charges are stored in batteries that are connected to the things providing said pseudo tools.

### Describe alternatives you have considered.

-Begrudgingly accepting that I can only tell my followers to craft 2 units of liquid ammonia at a time no matter how many very large storage batteries are connected to the grid powering the needed pseudo tool appliances

### Additional context

I asked if there was an actual lore or engineering-related reason for the arbitrary cap on [this reddit post](https://www.reddit.com/r/Cataclysm_DDA/comments/10zr55u/is_there_a_reason_for_why_vehicle_part_pseudo/), but since no one replied, I'm going to assume that there isn't one.

**Comment by AcidAntOnAMinefield** [comment]
Can confirm this is an issue:

- I cannot craft any batches above a certain amount (e.g.: milk, 16 units), despite having unlimited energy (both in my vehicle *and* in an Atomic Coffee Maker). Literally every recipe I want to batch craft I have to do it in batches of 16 because of whatever lead to this. 

I can't even craft 50 units of water if I tried to, the game limits me to 48. 

I assume this issue is related to https://github.com/CleverRaven/Cataclysm-DDA/issues/64275 which for whatever reason spiraled down into a debate about unrelated issues.

**Comment by irwiss** [comment]
The limits are not arbitrary - they correlate to the default battery size defined for the previously handheld on-batteries kind of tool.

In vanilla "atomic" coffee maker is no longer infinite liquid boiling tool, just a slightly more effective than normal coffee maker, the infinite part was moved to Aftershock

PRs like https://github.com/CleverRaven/Cataclysm-DDA/pull/64350 replace handheld tool definitions with "pseudo" ones that should be able to support larger charge amounts

**Comment by AmarinReyny** [comment]
> The limits are not arbitrary - they correlate to the default battery size defined for the previously handheld on-batteries kind of tool.
> 
> In vanilla "atomic" coffee maker is no longer infinite liquid boiling tool, just a slightly more effective than normal coffee maker, the infinite part was moved to Aftershock
> 
> PRs like #64350 replace handheld tool definitions with "pseudo" ones that should be able to support larger charge amounts

Ah, okay... Still, from an in-game perspective, it kind of is arbitrary, since tool is running off a vehicle/grid battery and not a handheld battery, and the total cap should depend on how much is in the vehicle/grid batteries, rather than the tool's usual handheld battery limitations - or on the limitations of an invisible pseudo battery, for that matter. That, and even for tools that have no handheld equivalent, it's still noticeable with recipes that have a ridiculous charge requirement that can easily disappoint/frustrate those who did the work necessary to set up the requirements for crafting (liquid ammonia comes to mind). Maybe the number for the max charges available in the pseudo battery should have some more zeroes in it?

**Comment by irwiss** [comment]
The tools that have pseudo battery do depend on how much battery charges are available in the vehicle/grid, the ones that don't likely weren't yet converted to use the pseudo battery

I'm not saying it shouldn't, I'm just explaining how it works since the cap isn't arbitrary - if tool "item" definition defaults to medium battery the cap is 500kJ, once they're converted to use pseudo batteries the cap is 1 million kJ

**Comment by AmarinReyny** [comment]
> The tools that have pseudo battery do depend on how many battery charges are available in the vehicle/grid, the ones that don't likely weren't yet converted to use the pseudo battery
> 
> I'm not saying it shouldn't, I'm just explaining how it works since the cap isn't arbitrary - if tool "item" definition defaults to medium battery the cap is 500kJ, once they're converted to use pseudo batteries the cap is 1 million kJ

So, I decided to go do some digging in the JSON to see what was going on, and what could be changed, regarding the pseudo items related to making liquid ammonia. First, I increased the pseudo battery capacity to 50 million, not 1 million (question: is there a reason for why it has both a capacity for the item itself, and an identical "battery" amount in its magazine?). I tested that in-game, and noticed that the charges available to use for the Ammonia Machine Reactor went up while looking at the recipe, but the same didn't happen for the rest.

I then did some file searching in Notepad++ to locate the appliance's pseudo tools themselves (question: is there a reason the pseudo tools are in items/tools/json's science.json file, and not in the same file as the other pseudo items?), and saw that the other relevant pseudo tools' magazines were not properly set up to get their charges from the pseudo battery; the nitrogen generator used only a list of storage batteries without listing the pseudo battery at all, and the other two had storage batteries listed alongside pseudo batteries, in all their magazine wells' item restrictions. I've changed that, and I am now trying to set things up in-game with even more batteries and mod-added free energy reactors than I previously had to see how this works going forward.
